pkg/semtech-loramac/contrib/semtech_loramac_timer.c

```diff
@@ -44,7 +44,7 @@ void TimerStart(TimerEvent_t *obj)
     xtimer_t *timer = &(obj->dev);
     msg_t *msg = &(obj->msg);
     msg->type = MSG_TYPE_MAC_TIMEOUT;
-    msg->content.ptr = obj->cb;
+    msg->content.ptr = (void*)(uintptr_t)obj->cb;
```

Comment:
Maybe using `msg->content.value` (which is of type `uint32_t`) would be a solution, as 32 bit should be wide enough for ROM addresses of constrained devices for quite some time. But I have the feeling that casting between `uint32_t` and function pointers is nothing the compiler will just accepted, when at least a modest level of warnings is enabled.

Maybe it is indeed better to have this discussion separately from this PR.